app/screens/referral-screen/referral-screen.tsx

```diff
@@ -0,0 +1,154 @@
+import { observer, inject } from "mobx-react"
+import * as React from "react"
+import { NavigationScreenProps, SafeAreaView } from "react-navigation"
+import {
+  Clipboard,
+  Share,
+  TextStyle,
+  View,
+  ViewStyle,
+} from "react-native"
+
+import { RootStore } from "../../models/root-store"
+import { UserStore } from "../../models/user-store"
+
+import { Button } from "../../components/button"
+import { Screen } from "../../components/screen"
+import { Sheet } from "../../components/sheet"
+import { Text } from "../../components/text"
+
+import { color, spacing } from "../../theme"
+
+import { logAnalyticsEvent } from "../../utils/analytics"
+
+export interface ReceiveScreenProps extends NavigationScreenProps {
+  userStore: UserStore,
+  appReferralLink: string
+}
+const ROOT: ViewStyle = {
+  flex: 1,
+  backgroundColor: color.primary,
+}
+const SCREEN: ViewStyle = {
+  flexGrow: 1,
+  alignItems: "center",
+  justifyContent: "center",
+}
+const HEADER_BAR: ViewStyle = {
+  alignItems: "flex-end",
+  width: "100%",
+  paddingHorizontal: spacing[4],
+  paddingVertical: spacing[1],
+}
+const INNER: ViewStyle = {
+  alignItems: "stretch",
+  padding: spacing[4],
+}
+const SHEET: ViewStyle = {
+  alignItems: "center",
+  maxWidth: 208,
+  marginVertical: spacing[5],
+  padding: spacing[5],
+  paddingBottom: spacing[0],
+  backgroundColor: color.palette.white,
+}
+const COPY_BUTTON: TextStyle = {
+  marginTop: spacing[3],
+  textAlign: "center",
+}
+const ADDRESS: TextStyle = {
+  marginTop: spacing[4],
+}
+const BOTTOM_BAR: ViewStyle = {
+  alignItems: "center",
+  width: "100%",
+  padding: spacing[3],
+  backgroundColor: color.palette.white,
+}
+
+@inject((rootStore: RootStore) => ({
+  userStore: rootStore.userStore,
+  appReferralLink: rootStore.userStore.userAppReferralLink,
```

Comment:
Seems injecting `appReferralLink` is redundant here as you can already retrieve it from userStore